shell/platform/windows/text_input_manager.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_TEXT_INPUT_MANAGER_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_TEXT_INPUT_MANAGER_H_
+
+#include <string>
+#include <Windows.h>
+#include <Windowsx.h>
+
+#include "flutter/shell/platform/common/cpp/geometry.h"
+
+namespace flutter {
+
+class TextInputManager {
+ public:
+  TextInputManager() noexcept = default;
+
+  void CreateImeWindow(HWND hwnd);
+  void DestroyImeWindow(HWND hwnd);
+  void UpdateImeWindow(HWND hwnd);
+
+  void UpdateCaretRect(HWND hwnd, const Rect& rect);
+
+  long GetComposingCursorPosition(HWND hwnd) const;
+  bool GetComposingString(HWND hwnd, std::u16string* result) const;
```

Comment:
std::optional here and in the next two methods.

Reply:
Done.